### DISCUSSION TITLE
Manual Navigation, Camera feed, two-way audio

### DISCUSSION THREAD

**Discussion by FabrizioRomanoGenovese** [Hypfer/Valetudo#1783] [Q&A / Support] [answered]
First of all, I wanted to thank you for this wonderful project!

I was thinking about buying a Dreame L10s Ultra but I was quite suspicious of the privacy implications, and this project saves it all.

This said, having done a bit of research, the L10s ultra has some nice features: The one I'm interested in at the moment is the following: through the DreameHome app, you can use the robot as a surveillance camera, as shown in this picture:
![pic](https://fs.npstatic.com/userfiles/7446224/image/dreame-l10s-ultra/NextPit-dreame-l10s-ultra-camera-w782.png)
As one can see, there are controls for navigation, there's camera feed, and two-way audio. This could be useful, for instance, to assist someone (e.g. an older person) when one isn't at home, or to monitor a pet.

I was wondering how much of this can be replicated using Valetudo.
- It seems that manual navigation is possible, as shown in [this project](https://gist.github.com/Hypfer/fcfa39996bd7522bbe2c5f18acb1fcf4), for instance. About this, my question is if manual navigation is exposed via MQTT or if I have to use HTTP put requests (judging from what it is said [here](https://github.com/Hypfer/Valetudo/discussions/1781) it seems like manual navigation won't be added to MQTT). In general, I wasn't able to find a completely documented HTTP API yet. 
- Do we have access to the front camera feed, for instance via RTSP? If so, does this feed include audio? This would provide vision and one-way audio. With these two points, I should already be able to hack together a primitive app to watch, listen and navigate.
- About two-way audio, I've seen people here experimenting with sending audio to the robot. This has been, for instance, discussed [here](https://github.com/Hypfer/Valetudo/issues/114). I don't know if someone experimented yet with continuous audio stream tho: e.g. while keeping a button pressed and speaking, the audio is directly played by the robot.

Sorry if some of these questions may be out of topic. I've just started diving into this word and I'm doing as much research as I can before buying a robot. Basically, I was wondering if someone already attempted to do what I want to do, and/or if someone could point me towards useful material to develop this project further.

**Comment by Hypfer** [comment, marked as answer]
Duplicate of https://github.com/Hypfer/Valetudo/discussions/1723

**Comment by FabrizioRomanoGenovese** [comment]
Makes sense. In this case I'd be inclined to buy a L10 Ultra and saving some money. I only find threads about the L10s ultra tho. Can I assume that the L10 ultra is also compatible with Valetudo without major differences?

**Reply by Hypfer** [reply]
No, you cannot assume that.
